HTML5-frontend/ndt-library.js

```diff
@@ -0,0 +1,234 @@
+/**
+ * @fileoverview This javascript code includes an NDT object that contains the
+ * methods that a customized front end implemetation of the NDT test could
+ * utilize.  The NDT object is self invoked when the library is included on a
+ * page, however, the frontend can customize certain properties when the
+ * interaction is initialized with the startTest method.
+ *
+ * Dependencies: ndt-wrapper.js, ndt-browser-client.js
+ */
+
+
+/**
+ * Immediately invoked function expression that creates NDT library object
+ */
+(function (window) {
+  'use strict';
+
+  /** @constructor */
+  function NDTLibjs() {
+
+    // Look up a suitable NDT server using mlab-ns. The ndtServer property is
+    //  used by the backend javascript to determine which server to test
+    //  against.
+    this.getServer();
+
+    // Initializes simulate property to false.  Can be changed upon starting a
+    // test in the testServer method
+    this.simulate = false;
+
+    // STATUS VARIABLES
+    this.use_websocket_client = false;
+    this.websocket_client = null;
+
+  }
+
+  /**
+   * Operates on an instance of NDT to begin executing a test.
+   * @param - none
+   * @return - {string} True if client is using websockets, False if error
+   * occurred during finding backend server, or applet container element fall
+   * back to Java applet.
+   */
+  NDTLibjs.prototype.startTest = function() {
+    if (NDT.ndtServer) {
```

Comment:
I'm not sure that the class should expect NDT to be a window properties rather than an object passed in.